source/data-transfer.js

```diff
@@ -2280,11 +2280,20 @@ Skylink.prototype._handleDataTransferTimeoutForPeer = function (transferId, peer
  */
 Skylink.prototype._processDataChannelData = function(rawData, peerId, channelName, channelType) {
   var self = this;
-
-  var channelProp = channelType === self.DATA_CHANNEL_TYPE.MESSAGING ? 'main' : channelName;
-  var transferId = self._dataChannels[peerId][channelProp].transferId || null;
-  var streamId = self._dataChannels[peerId][channelProp].streamId || null;
+  var transferId = null;
+  var streamId = null;
   var isStreamChunk = false;
+  var channelProp = channelType === self.DATA_CHANNEL_TYPE.MESSAGING ? 'main' : channelName;
+
+  // Safe access of _dataChannel object in case dataChannel has been closed unexpectedly | ESS-983
+  var objPeerDataChannel = self._dataChannels[peerId] || {};
+  if (objPeerDataChannel.hasOwnProperty(channelProp) && typeof objPeerDataChannel[channelProp] === 'object') {
+    transferId = objPeerDataChannel[channelProp].transferId;
+    streamId = objPeerDataChannel[channelProp].streamId;
+  }
```

Comment:
please commit files only which you have worked, not auto generated files.

Reply:
@geekavi I've reverted the auto generated files and pushed them. Please check now if this is okay?